pkg/storage/unified/apistore/prepare.go

```diff
@@ -24,12 +24,13 @@ func (s *Storage) prepareObjectForStorage(ctx context.Context, newObject runtime
 	if err != nil {
 		return nil, err
 	}
-	if obj.GetName() == "" {
-		return nil, fmt.Errorf("new object must have a name")
+	if obj.GetName() == "" && obj.GetGenerateName() == "" {
```

Comment:
```suggestion
	if obj.GetName() == "" {
```
The generate name input should have already been processed to set the name

Reply:
so, `generateName` it's a thing populated by the k8s server?
(changing it, I'm just curious how this works)